framework/src/libtorch/utils/LibtorchArtificialNeuralNet.C

```diff
@@ -205,8 +222,16 @@ dataLoad<Moose::LibtorchArtificialNeuralNet>(
   activation_functions.resize(num_activation_items);
   dataLoad(stream, activation_functions, context);
 
+  int8_t device_type;
+  dataLoad(stream, device_type, context);
+  const torch::DeviceType divt(static_cast<torch::DeviceType>(device_type));
+
+  int8_t data_type;
```

Comment:
Can you use `std::underlying_type<torch::DeviceType>::type` instead of `int8_t`? Also interesting that you don't need a `typename` in front of `std::underlying_type<torch::DeviceType>::type` 🤔

Reply:
Added the mods!